app/src/lib/app-state.ts

```diff
@@ -695,6 +695,17 @@ export interface ICompareState {
   }
 }
 
+export interface IDivergingBranchBannerState {
+  /** Show the diverging notification banner */
+  readonly isPromptVisible: boolean
+
+  /** Has the user dismissed the notification banner? */
+  readonly isPromptDismissed: boolean
+
+  /** Show the diverging notification nudge on the tab */
+  readonly isNudgeVisible: boolean
+}
```

Comment:
I've decided two put these 3 flags under the same object rather than having 3 separate flags starting with `divergingBranchBannerXXXX`. This also makes it slightly easier to update the state from the `app-store`.